File: CourierAmexClient/src/app/features/package/models/package-item.model.ts
import { BaseEntity } from "@app/models";

export interface PackageItemModel extends BaseEntity<number> {
    number: number; // NUMERO
    brandId: string; // ID_MARCA
    modelId: string; // ID_MODELO
    series?: string; // SERIE
    characteristics: string; // CARACTERISTICAS
    description: string; // DESCRIPCION
    composition?: string; // COMPOSICION
    quantity: number; // CANTIDAD
    unitCost: number; // COSTO_UNITARIO
    origin: string; // PROCEDENCIA
    source: string; // ORIGEN
    state: string; // ESTADO
    style?: string; // ESTILO
    color?: string; // COLOR
    size?: string; // TALLA
    batch?: string; // PARTIDA
    invoice?: string; // FACTURA
    inclusionDate: Date; // FEC_INCLUSION
    inclusionUser: number; // USER_INCLUSION
    modificationDate?: Date; // FEC_MODIFICA
    modificationUser?: number; // USER_MODIFICA
    invoiceDate?: Date; // FECHA_FACTURA
    cost?: number; // COSTO
}

export let NewPackageItemModel : PackageItemModel = {
    id: 0, // Assuming BaseEntity<number> requires an ID
    number: 0, // Default number
    brandId: '', // Empty string for required fields
    modelId: '',
    series: undefined, // Optional field
    characteristics: '',
    description: '',
    composition: undefined,
    quantity: 1, // Default to 1
    unitCost: 0.0,
    origin: '',
    source: '',
    state: '', // Default state
    style: undefined,
    color: undefined,
    size: undefined,
    batch: undefined,
    invoice: undefined,
    inclusionDate: new Date(), // Set current date
    inclusionUser: 0,
    modificationDate: undefined,
    modificationUser: undefined,
    invoiceDate: undefined,
    cost: undefined,
    status: 0
};


export class PackageItem_PreviousReport {
    id!: number;
    fullName!: string;
    packageNumber!: number;
    bag!: string;
    packageDescription!: string;
    brand!: string;
    model!: string;
    serialNumber!: string;
    description!: string;
    composition!: string;
    quantity!: number;
    unitCost!: number;
    origin!: string;
    status!: string;
    style!: string;
    color!: string;
    size!: string;
    itemNumber!: string;
    invoice!: string;
    source!: string;
    totalPrice!: number;
    packagePrice!: number;
    characteristics!: string;
    invoiceDate!: Date | null;
    totalRows!: number;
  }
  
  export const newPackageItem_PreviousReport: PackageItem_PreviousReport = {
    id: 0,
    fullName: '',
    packageNumber: 0,
    bag: '',
    packageDescription: '',
    brand: '',
    model: '',
    serialNumber: '',
    description: '',
    composition: '',
    quantity: 0,
    unitCost: 0,
    origin: '',
    status: '',
    style: '',
    color: '',
    size: '',
    itemNumber: '',
    invoice: '',
    source: '',
    totalPrice: 0,
    packagePrice: 0,
    characteristics: '',
    invoiceDate: null,
    totalRows: 0,
  };

  
  

File: CourierAmexAPI/CourierAmex/Models/PackageItemModel.cs
using CourierAmex.Storage.Domain;

namespace CourierAmex.Models
{
    public class PackageItemModel : BaseEntity<int>
    {
        public int Number { get; set; } // NUMERO
        public required string BrandId { get; set; } // ID_MARCA
        public required string ModelId { get; set; } // ID_MODELO
        public string? Series { get; set; } // SERIE
        public string Characteristics { get; set; } // CARACTERISTICAS
        public required string Description { get; set; } // DESCRIPCION
        public string? Composition { get; set; } // COMPOSICION
        public required int Quantity { get; set; } // CANTIDAD
        public required decimal UnitCost { get; set; } // COSTO_UNITARIO
        public required string Origin { get; set; } // PROCEDENCIA
        public required string Source { get; set; } // ORIGEN
        public required string State { get; set; } // ESTADO
        public string? Style { get; set; } // ESTILO
        public string? Color { get; set; } // COLOR
        public string? Size { get; set; } // TALLA
        public string? Batch { get; set; } // PARTIDA
        public string? Invoice { get; set; } // FACTURA
        public DateTime InclusionDate { get; set; } // FEC_INCLUSION
        public int InclusionUser { get; set; } // USER_INCLUSION
        public DateTime? ModificationDate { get; set; } // FEC_MODIFICA
        public int? ModificationUser { get; set; } // USER_MODIFICA
        public DateTime? InvoiceDate { get; set; } // FECHA_FACTURA
        public decimal? Cost { get; set; } // COSTO
    }

    //public class PackageItemModel_PreviousReport_PreStudy
    //{
    //    public int Id { get; set; }
    //    public string FullName { get; set; }
    //    public int PackageNumber { get; set; }
    //    public string Bag { get; set; }
    //    public string PackageDescription { get; set; }
    //    public string Brand { get; set; }
    //    public string Model { get; set; }
    //    public string SerialNumber { get; set; }
    //    public string Description { get; set; }
    //    public string Composition { get; set; }
    //    public int Quantity { get; set; }
    //    public decimal UnitCost { get; set; }
    //    public string Origin { get; set; }
    //    public string Status { get; set; }
    //    public string Style { get; set; }
    //    public string Color { get; set; }
    //    public string Size { get; set; }
    //    public string ItemNumber { get; set; }
    //    public string Invoice { get; set; }
    //    public string Source { get; set; }
    //    public decimal TotalPrice { get; set; }
    //    public decimal PackagePrice { get; set; }
    //    public string Characteristics { get; set; }
    //    public DateTime? InvoiceDate { get; set; }
    //    public int TotalRows { get; set; }
    //}
}
